Initial application + testing refactor for cnetbin2pvl

AustinSanders opened the pull request:
Refactored cnetbin2pvl application into .cpp, .h, and main.cpp files.  Adds tests for cnetbin2pvl.

## Description
Converts the cnetbin2pvl application into the new format by separating it into .cpp and .h files.  Also adds a test for cnetbin2pvl.  The test simply writes / reads the pvl file.

## Related Issue
[here](https://trello.com/c/Kj9DyCsz)

## Motivation and Context
Update the ISIS api and reduce testing overhead.

## How Has This Been Tested?
Changes were tested using the included gtest.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- - [ ] My code follows the code style of this project. -->
- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

AustinSanders committed:
Initial application/testing refactor of cnetbin2pvl

AustinSanders committed:
Removed makefile for old test.

AustinSanders committed:
Adjusted spacing to better comply with isis3 internal standard.

jessemapel commented on isis/src/control/apps/cnetbin2pvl/cnetbin2pvl.cpp:
```diff
@@ -0,0 +1,23 @@
+
+namespace Isis{
+  void cnetbin2pvl(UserInterface &ui) {
+    // Build a histogram from the control net
```
Remove this comment

jessemapel commented on isis/src/control/apps/cnetbin2pvl/cnetbin2pvl.h:
```diff
@@ -0,0 +1,13 @@
+
+namespace Isis {
+  extern void cnetbin2pvl(UserInterface &ui);
+  extern void cnetbin2pvl(ControlNet &net, Progress &progress, UserInterface &ui);
```
Progress should be a pointer that defaults to nullptr so that it's not used when this is called outside of the app.

jessemapel commented on isis/tests/cnetbin2pvlTests.cpp:
```diff
@@ -0,0 +1,41 @@
+  FILE* cnetFile;
+  cnetFile = fopen(cnetFilePath.toStdString().c_str(), "wb");
+  fwrite(network, sizeof(char), sizeof(*network), cnetFile);
+  fclose(cnetFile);
```
This isn't how binary networks are written out. They have a specific on disk binary format. Use ControlNet::Write with pvl=false to write out the network as binary.

AustinSanders replied:
I definitely made a mistake here.  The "from" parameter from the UI requires a path to a control network, but the pvl is written based on the "network" parameter that's passed to the function. It doesn't look like "from" is used at all, so there's no need to write anything to disk.

jessemapel reviewed (changes requested):
I don't think your test will work due to how it generates a binary network.

AustinSanders committed:
Merge branch 'dev' of https://github.com/USGS-Astrogeology/ISIS3 into testing

AustinSanders committed:
Removed default cnetbin2pvl testing makefile

AustinSanders commented on isis/tests/cnetbin2pvlTests.cpp:
```diff
@@ -0,0 +1,41 @@
+
+  Pvl pvl;
+  try {
+    pvl.read(cnetFilePath);
```
Should be reading pvlOut, not cnetFilePath

AustinSanders committed:
Addressed PR comments / reviews.

Removed old comment
Removed unnecessary cnet write
Added default null pointer for progress parameter.

AustinSanders committed:
Removed extra whitespace

jessemapel commented on isis/tests/cnetbin2pvlTests.cpp:
```diff
@@ -0,0 +1,35 @@
+
+  Pvl pvl;
+  try {
+    pvl.read(pvlOut);
```
Binary control networks have a PVL formatted header, so this will still pass even if it writes out a binary network.

You should add an additional check that the PVL contains an object called ControlPoint

jessemapel reviewed (approved):
Looks good to me.
